### ISSUE TITLE
98% accuracy code in previous tensorflow version giving huge loss and very less accuracy in this version. 

### ISSUE THREAD

**Issue by aryan5276** [apple/tensorflow_macos#247]
I have a code that creates models of 98-99% accuracy and 0.2-0.3 loss when used with previous versions of tensorflow on windows computer. On running the same code on my macbook air m1 using tensorflow 2.4.0-rc0, it showed huge losses and very less accuracy. I am getting no errors but every combination of layers I try gives me huge losses. Also, even though i get accuracy of 70-80% while training, on testing it drops down to 40-45% and the loss surges to a 6 digit number on the final test. Please provide the solution to this issue.

Here is my code: 

import tensorflow as tf
import cv2
import numpy as np
import os
import sys

from sklearn.model_selection import train_test_split

EPOCHS = 10
IMG_WIDTH = 30
IMG_HEIGHT = 30
NUM_CATEGORIES = 43
TEST_SIZE = 0.4

def main():

    # Check command-line arguments
    if len(sys.argv) not in [2, 3]:
        sys.exit("Usage: python traffic.py data_directory [model.h5]")

    # Get image arrays and labels for all image files
    images, labels = load_data(sys.argv[1])

    # Split data into training and testing sets
    labels = tf.keras.utils.to_categorical(labels)
    x_train, x_test, y_train, y_test = train_test_split(
        np.array(images), np.array(labels), test_size=TEST_SIZE
    )

    # Get a compiled neural network
    model = get_model()

    # Fit model on training data
    model.fit(x_train, y_train, epochs=EPOCHS)

    # Evaluate neural network performance
    model.evaluate(x_test,  y_test, verbose=2)

    # Save model to file
    if len(sys.argv) == 3:
        filename = sys.argv[2]
        model.save(filename)
        print(f"Model saved to {filename}.")

def load_data(data_dir):
    """
    Load image data from directory `data_dir`.

    Assume `data_dir` has one directory named after each category, numbered
    0 through NUM_CATEGORIES - 1. Inside each category directory will be some
    number of image files.

    Return tuple `(images, labels)`. `images` should be a list of all
    of the images in the data directory, where each image is formatted as a
    numpy ndarray with dimensions IMG_WIDTH x IMG_HEIGHT x 3. `labels` should
    be a list of integer labels, representing the categories for each of the
    corresponding `images`.
    """
    images = []
    labels = []
    for folder in os.listdir(data_dir):
        folder_path = os.path.join(data_dir, folder)
        if os.path.isdir(folder_path):
            for image_file in os.listdir(folder_path):
                image = cv2.imread(os.path.join(folder_path, image_file), cv2.IMREAD_COLOR)
                image = cv2.resize(image, (IMG_WIDTH, IMG_HEIGHT), interpolation=cv2.INTER_AREA)
                images.append(image)
                labels.append(int(folder))

    return images, labels

def get_model():
    """
    Returns a compiled convolutional neural network model. Assume that the
    `input_shape` of the first layer is `(IMG_WIDTH, IMG_HEIGHT, 3)`.
    The output layer should have `NUM_CATEGORIES` units, one for each category.
    """
    # Create a convolutional neural network
    model = tf.keras.models.Sequential([

        # Convolutional layer.
        tf.keras.layers.Conv2D(
            32, (3, 3), activation="relu", input_shape=(IMG_WIDTH, IMG_HEIGHT, 3)    # diff nos and size of filter
        ),

        # Max-pooling layer, using 2x2 pool size
        tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),  # diff pool sizes

        tf.keras.layers.Conv2D(
            64, (3, 3), activation="relu"  # diff nos and size of filter
        ),

        tf.keras.layers.MaxPooling2D(pool_size=(2, 2)),

        # Flatten units
        tf.keras.layers.Flatten(),

        # Add a hidden layer with dropout
        tf.keras.layers.Dense(128, activation="relu"),
        tf.keras.layers.Dense(64, activation="relu"),
        tf.keras.layers.Dropout(0.33),                           # diff values of dropout

        # Add an output layer with output units for all 43 categories
        tf.keras.layers.Dense(NUM_CATEGORIES, activation="softmax")
    ])

    # Train neural network
    model.compile(
        optimizer="adam",
        loss="categorical_crossentropy",
        metrics=["accuracy"]
    )

    return model

if __name__ == "__main__":
    main()

Output: 
2021-04-29 13:00:30.215162: I tensorflow/compiler/mlir/mlir_graph_optimization_pass.cc:116] None of the MLIR optimization passes are enabled (registered 2)
2021-04-29 13:00:30.215309: W tensorflow/core/platform/profile_utils/cpu_utils.cc:126] Failed to get CPU frequency: 0 Hz
Epoch 1/10
500/500 [==============================] - 3s 6ms/step - loss: 7.4314 - accuracy: 0.1835
Epoch 2/10
500/500 [==============================] - 3s 6ms/step - loss: 30.3907 - accuracy: 0.5086
Epoch 3/10
500/500 [==============================] - 3s 6ms/step - loss: 101.8741 - accuracy: 0.6245
Epoch 4/10
500/500 [==============================] - 3s 6ms/step - loss: 290.0572 - accuracy: 0.6668
Epoch 5/10
500/500 [==============================] - 3s 5ms/step - loss: 711.0809 - accuracy: 0.7014
Epoch 6/10
500/500 [==============================] - 3s 5ms/step - loss: 1029.0363 - accuracy: 0.7431
Epoch 7/10
500/500 [==============================] - 3s 5ms/step - loss: 1491.8649 - accuracy: 0.7616
Epoch 8/10
500/500 [==============================] - 3s 5ms/step - loss: 2219.3415 - accuracy: 0.7707
Epoch 9/10
500/500 [==============================] - 3s 5ms/step - loss: 4306.0616 - accuracy: 0.7636
Epoch 10/10
500/500 [==============================] - 3s 5ms/step - loss: 5144.7086 - accuracy: 0.7782
333/333 - 1s - loss: 355651.9062 - accuracy: 0.4495

Process finished with exit code 0

**Comment by Darky2020** [comment]
I have a similar problem with my DCGAN. It seems to give a decent loss in google colaboratory but results in huge losses when I train it.

My code: https://pastebin.com/tDKDC85d
